src/Message/UserMessage.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Message;
+
+use PhpLlm\LlmChain\Message\Content\ContentInterface;
+use PhpLlm\LlmChain\Message\Content\Image;
+use PhpLlm\LlmChain\Message\Content\Text;
+
+final readonly class UserMessage implements MessageInterface
+{
+    /**
+     * @var list<ContentInterface>
+     */
+    public array $content;
+
+    public function __construct(
+        ContentInterface ...$content,
+    ) {
+        $this->content = $content;
+    }
+
+    public function getRole(): Role
+    {
+        return Role::User;
+    }
+
+    /**
+     * @return array{
+     *     role: Role::User,
+     *     content: string|list<array{type: 'text', text: string}|array{type: 'image_url', image_url: array{url: string}}>
+     * }
+     */
+    public function jsonSerialize(): array
+    {
+        $array = ['role' => Role::User];
+        if (1 === count($this->content) && $this->content[0] instanceof Text) {
+            $array['content'] = $this->content[0]->text;
+
+            return $array;
+        }
+
+        foreach ($this->content as $entry) {
+            $array['content'][] = $entry->jsonSerialize();
+        }
```

Comment:
should be sufficient to just call
```suggestion
        $array['content'] = $this->content;
```

Reply:
Nice. Thanks. Did not worked with this JsonSerializable interface a lot in the past. Cool feature! 👍

Reply:
But! What to mention here. This does only work when utilizing `json_encode` it will not automatically parse it to arrays. So i have to adjust the tests to test encoded arrays instead of comparing the arrays. 

But for me it is fine. The code must surely not be different just to utilize the array instead of the json encoded array.